picard/cluster.py

```diff
@@ -315,8 +316,14 @@ def cluster(files):
 
         cluster_list = defaultdict(FileCluster)
         for file in files:
-            artist = file.metadata['albumartist'] or file.metadata['artist']
-            album = file.metadata['album']
+            metadata = file.metadata
+            # If the file is attached to a track we should use the original
+            # metadata for clustering. This is often used by users when moving
+            # mismatched files back from the right pane to the left.
+            if isinstance(file.parent, Track):
```

Comment:
Rather than setting and then re-setting metadata, wouldn't it be better to do an else instead?

Reply:
I changed it to if-else